skimage/future/detect_obj/cascade.pyx

```diff
@@ -0,0 +1,832 @@
+                                                           detections[current_detection])
+
+    clusters = threshold_clusters(clusters, min_neighbour_amount)
+    return get_mean_detections(clusters)
+
+
+cdef DetectionsCluster update_cluster(DetectionsCluster cluster,
+                                      Detection detection):
+    """Updated the cluster by adding new detection.
+
+    Updates the cluster by adding new detection to it. The added
+    detection contributes to the mean values of the cluster.
+
+    Parameters
+    ----------
+    cluster : DetectionsCluster
+        A cluster of detections.
+    detection : Detection
+        The detection to be added to cluster.
+
+    Returns
+    -------
+    updated_cluster : DetectionsCluster
+        The updated cluster.
+    """
+
+    cdef DetectionsCluster updated_cluster = cluster
+
+    updated_cluster.r_sum += detection.r
+    updated_cluster.c_sum += detection.c
+    updated_cluster.width_sum += detection.width
+    updated_cluster.height_sum += detection.height
+    updated_cluster.count += 1
+
+    return updated_cluster
+
+
+cdef Detection mean_detection_from_cluster(DetectionsCluster cluster):
+    """Compute the mean detection from the cluster.
+
+    Returns the mean detection computed from the all rectangles that
+    belong to current cluster.
+
+    Parameters
+    ----------
+    cluster : DetectionsCluster
+        A cluster of detections.
+
+    Returns
+    -------
+    mean : Detection
+        The mean detection.
+    """
+
+    cdef Detection mean
+
+    mean.r = cluster.r_sum / cluster.count
+    mean.c = cluster.c_sum / cluster.count
+    mean.width = cluster.width_sum / cluster.count
+    mean.height = cluster.height_sum / cluster.count
+
+    return mean
+
+
+cdef DetectionsCluster cluster_from_detection(Detection detection):
+    """Create a cluster from a single detection.
+
+    Creates a cluster with count one and values that are taken from detection.
+
+    Parameters
+    ----------
+    detection : Detection
+        A single detection.
+
+    Returns
+    -------
+    new_cluster : DetectionsCluster
+        The cluster struct that was created from detection.
+    """
+
+    cdef DetectionsCluster new_cluster
+
+    new_cluster.r_sum = detection.r
+    new_cluster.c_sum = detection.c
+    new_cluster.width_sum = detection.width
+    new_cluster.height_sum = detection.height
+    new_cluster.count = 1
+
+    return new_cluster
+
+
+cdef vector[DetectionsCluster] threshold_clusters(vector[DetectionsCluster] clusters,
+                                                  int count_threshold):
+    """Threshold clusters depending on the amount of rectangles in them.
+
+    Only the clusters with the amount of rectangles greater than the threshold
+    are left.
+
+    Parameters
+    ----------
+    clusters : vector[DetectionsCluster]
+        Array of rectnagles clusters.
+    count_threshold : int
+        The threshold amount of rectangles that is used.
```

Comment:
``amount`` is reserved for uncountables such as "sugar", "water",etc.  For things you can count use "number of".